static/js/plotjs/stackm.js

```diff
@@ -3,146 +3,203 @@ const binsize = slider.value;
-
-  binsource.data.mjd = [];
-  binsource.data.flux = [];
-  binsource.data.fluxerr = [];
-  binsource.data.filter = [];
-  binsource.data.color = [];
-  binsource.data.lim_mag = [];
-  binsource.data.mag = [];
-  binsource.data.magerr = [];
-  binsource.data.instrument = [];
-  binsource.data.stacked = [];
-
-  binerrsource.data.xs = [];
-  binerrsource.data.ys = [];
-  binerrsource.data.color = [];
-
-  unobsbinsource.data.mjd = [];
-  unobsbinsource.data.lim_mag = [];
-  unobsbinsource.data.color = [];
-  unobsbinsource.data.flux = [];
-  unobsbinsource.data.fluxerr = [];
-  unobsbinsource.data.mag = [];
-  unobsbinsource.data.magerr = [];
-  unobsbinsource.data.filter = [];
-  unobsbinsource.data.stacked = [];
-  unobsbinsource.data.instrument = [];
-
-  for (var j = 0; j < fluxsource.get_length(); j++) {
-    fluxsource.data.alpha[j] = fluxalph;
-    fluxerrsource.data.alpha[j] = fluxalph;
-  }
-
-  for (var j = 0; j < unobssource.get_length(); j++) {
-    if (isFinite(unobssource.data.flux[j])){
-      unobssource.data.alpha[j] = fluxalph;
-    }
-  }
+    const fluxsource = eval(`obs${i}`).data_source;
+    const binsource = eval(`bin${i}`).data_source;
+
+    const fluxerrsource = eval(`obserr${i}`).data_source;
+    const binerrsource = eval(`binerr${i}`).data_source;
+
+    const unobssource = eval(`unobs${i}`).data_source;
+    const unobsbinsource = eval(`unobsbin${i}`).data_source;
+
+    const boldsource = eval(`bold${i}`);
+    const allsource = eval(`all${i}`);
+
+    const minmjd = Math.min.apply(Math, fluxsource.data.mjd) - 15;
+    const maxmjd = Math.max.apply(Math, fluxsource.data.mjd) + 15;
+
+    boldsource.data.flux = [];
+    boldsource.data.fluxerr = [];
+    boldsource.data.mjd = [];
+    boldsource.data.filter = [];
+    boldsource.data.mag = [];
+    boldsource.data.magerr = [];
+    boldsource.data.lim_mag = [];
+    boldsource.data.zp = [];
+    boldsource.data.zpsys = [];
+    boldsource.data.stacked = [];
+
+    binsource.data.mjd = [];
+    binsource.data.flux = [];
+    binsource.data.fluxerr = [];
+    binsource.data.filter = [];
+    binsource.data.color = [];
+    binsource.data.lim_mag = [];
+    binsource.data.mag = [];
+    binsource.data.magerr = [];
+    binsource.data.instrument = [];
+    binsource.data.stacked = [];
+
+    binerrsource.data.xs = [];
+    binerrsource.data.ys = [];
+    binerrsource.data.color = [];
+
+    unobsbinsource.data.mjd = [];
+    unobsbinsource.data.lim_mag = [];
+    unobsbinsource.data.color = [];
+    unobsbinsource.data.flux = [];
+    unobsbinsource.data.fluxerr = [];
+    unobsbinsource.data.mag = [];
+    unobsbinsource.data.magerr = [];
+    unobsbinsource.data.filter = [];
+    unobsbinsource.data.stacked = [];
+    unobsbinsource.data.instrument = [];
+
+
+    if (binsize === 0) {
+        for (var j = 0; j < allsource.get_length(); j++) {
+            boldsource.data.flux.push(allsource.data.flux[j]);
```

Comment:
This can be simplified by iterating over the list of relevant keys (fine leaving as is for now). The other section below, as well.